src/core/ProjectRenderer.cpp

```diff
@@ -234,12 +234,12 @@ void ProjectRenderer::updateConsoleProgress()
 	prog[cols] = 0;
 
 	const auto activity = (const char*)"|/-\\";
-	memset( buf, 0, sizeof( buf ) );
-	sprintf( buf, "\r|%s|    %3d%%   %c  ", prog, m_progress,
+	memset(buf.data(), 0, sizeof(buf));
```

Comment:
I think `.size()` would be better than `sizeof` here (replacing `memcpy` with `std::fill` would be even better, but you might consider that to be out of scope).